### PR TITLE
added state_dict/load_state_dict for ReduceLROnPlateau

### PR DESCRIPTION
**Pull request by kahne**
There is a merged PR #5300 trying to make ReduceLROnPlateau serializable and resolve issue #5218. However, it's still not working. Moreover, it's more consistent to provide the common state_dict/load_state_dict interface for serialization.

### CODE REVIEW

**Comment by ezyang** [comment]
Thanks @kahne. Is there a test we can add for this?

**Comment by kahne** [comment]
@ezyang I tested it manually with `torch.save()` and `torch.load()`. Is there any existing tests for  lr scheduler that I can refer to? Thanks!

File: torch/optim/lr_scheduler.py [hunk 1]
```diff
@@ -374,3 +374,10 @@ def _init_is_better(self, mode, threshold, threshold_mode):
             self.mode_worse = (-float('inf'))
 
         self.is_better = partial(self._cmp, mode, threshold_mode, threshold)
+
+    def state_dict(self):
+        return {key: value for key, value in self.__dict__.items() if key not in {'optimizer', 'is_better'}}
```
Comment (apaszke) [inline torch/optim/lr_scheduler.py]: Can we please avoid reimplementing `state_dict` and do it like this instead?
```python
def state_dict(self):
    state_dict = super(ReduceLROnPlateau, self).state_dict()
    del state_dict['is_better']
    return state_dict
```

**Comment by apaszke** [comment]
@kahne there are a few tests in `test/test_optim.py`

File: torch/optim/lr_scheduler.py (hunk 1 above)
Comment (kahne) [reply torch/optim/lr_scheduler.py]: `ReduceLROnPlateau ` is not inherited from `_LRScheduler` (`object` instead). So I think it's not a reimplementation?

Comment (apaszke) [reply torch/optim/lr_scheduler.py]: Oh I see. Let's leave it as is then!

**Comment by kahne** [comment]
@ezyang @apaszke tests added

**Review by apaszke — APPROVED** [review]

**Comment by apaszke** [comment]
Thanks @kahne!

**Comment by ignaciorlando** [comment]
Hi!

I'm trying to use state_dict() on ReduceLROnPlateau but I got this error:
AttributeError: 'ReduceLROnPlateau' object has no attribute 'state_dict'

I'm using pytorch 0.4.0.

Thanks!

**Comment by kahne** [comment]
@ignaciorlando Hi! This feature is available in 0.4.1 but not 0.4.0. Can you upgrade to 0.4.1 and try again?

**Pull request merged**